typing/env.ml

```diff
@@ -3075,9 +3081,17 @@ let report_lookup_error _loc env ppf = function
   | Unbound_class lid ->
       fprintf ppf "Unbound class %a" !print_longident lid;
       spellcheck ppf extract_classes env lid;
-  | Unbound_modtype lid ->
+  | Unbound_modtype lid -> begin
       fprintf ppf "Unbound module type %a" !print_longident lid;
       spellcheck ppf extract_modtypes env lid;
+      match find_namesake_module lid env with
```

Comment:
I don't find the `find_namesake_module` auxiliary function very helpful, in part because "namesake" is a term I am not so familiar with. You could just call `find_module_by_name` directly. (You can use `match ... with exception Not_found -> ... | _ -> ...` if it leads to nicer code than `try .. with ..`.)